### PR TITLE
feat: antd auth page

### PR DESCRIPTION
**Pull request by biskuvit**
Added `<AuthPage>` for **Ant Design** package of **refine**

### Self Check before Merge

Please check all items below before reviewing.

-   [x] Corresponding issues are created/updated or not needed
-   [ ] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: .changeset/mean-ways-buy.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@pankod/refine-antd": patch
```
Comment (omeraplak) [inline .changeset/mean-ways-buy.md:2]:
```suggestion
"@pankod/refine-antd": minor
```

File: .changeset/wild-keys-tickle.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+```tsx
+type RefineAuthPageProps = {
+    type?: "login" | "register" | "resetPassword" | "updatePassword";
+    providers?: IProvider[];
+    submitButton?: React.ReactNode;
+    registerLink?: React.ReactNode;
+    loginLink?: React.ReactNode;
+    resetPasswordLink?: React.ReactNode;
+    updatePasswordLink?: React.ReactNode;
+    remember?: boolean;
```
Comment (omeraplak) [inline .changeset/wild-keys-tickle.md:18]:
```suggestion
    rememberMe?: boolean;
```
Could be ReactNode and false

**Review by omeraplak — APPROVED** [review]

**Review by yildirayunlu — APPROVED** [review]

File: packages/antd/src/components/pages/auth/components/register/index.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            >
+                <Form.Item
+                    name="email"
+                    label={translate("pages.register.email", "Email")}
+                    rules={[
+                        { required: true },
+                        {
+                            type: "email",
+                            message: translate(
+                                "pages.register.valiEmail",
```
Comment (omeraplak) [inline packages/antd/src/components/pages/auth/components/register/index.tsx:75]:
```suggestion
                                "pages.register.validEmail",
```
Let's ask the error text to the core team

File: .changeset/wild-keys-tickle.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ type IProvider = {
         TContentProps extends {} = Record<keyof any, unknown>,
     > = PropsWithChildren<{
         providers?: IProvider[];
-        submitButton?: React.ReactNode;
+        onSubmit?: (formValues: any) => void;
```
Comment (omeraplak) [inline .changeset/wild-keys-tickle.md:55]: Why did we use "any" type for formValues?

File: packages/antd/src/components/pages/auth/components/login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            {renderProviders()}
+            <Form<ILoginForm>
+                layout="vertical"
+                form={form}
+                onFinish={(values) => {
+                    if (onSubmit) {
+                        onSubmit(values);
+                    } else {
+                        login(values);
+                    }
```
Comment (omeraplak) [inline packages/antd/src/components/pages/auth/components/login/index.tsx:108]:
```suggestion
                    if (onSubmit) {
                        onSubmit(values);
                        return;
                    }
                    login(values);
                    
```

File: packages/antd/src/components/pages/auth/index.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                return <LoginPage {...props} />;
+            case "register":
+                return <RegisterPage {...props} />;
+            case "resetPassword":
+                return <ResetPasswordPage {...props} />;
+            case "updatePassword":
+                return <UpdatePasswordPage {...props} />;
+            default:
+                return <LoginPage {...props} />;
+        }
```
Comment (ozkalai) [inline packages/antd/src/components/pages/auth/index.tsx:33]:
```suggestion
        switch (type) {
            case "register":
                return <RegisterPage {...props} />;
            case "resetPassword":
                return <ResetPasswordPage {...props} />;
            case "updatePassword":
                return <UpdatePasswordPage {...props} />;
            default:
                return <LoginPage {...props} />;
        }
```

**Review by ozkalai — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

File: examples/fineFoods/admin/antd/src/authProvider.ts [hunk 7]
```diff
@@ -1,11 +1,40 @@
...
-        return Promise.resolve();
+    login: async ({ email, password }) => {
+        if (email && password) {
+            localStorage.setItem(TOKEN_KEY, `${email}-${password}`);
+            return Promise.resolve();
+        }
+        return Promise.reject("Invalid email or password");
+    },
+    register: async ({ email, password }) => {
+        if (!(email && password)) return Promise.reject();
```
Comment (omeraplak) [inline examples/fineFoods/admin/antd/src/authProvider.ts:15]:
```suggestion
```

File: examples/fineFoods/admin/antd/src/authProvider.ts [hunk 8]
```diff
@@ -1,11 +1,40 @@
...
 
 export const authProvider: AuthProvider = {
-    login: async ({ username, password }) => {
-        localStorage.setItem(TOKEN_KEY, `${username}-${password}`);
-        return Promise.resolve();
+    login: async ({ email, password }) => {
+        if (email && password) {
+            localStorage.setItem(TOKEN_KEY, `${email}-${password}`);
+            return Promise.resolve();
+        }
```
Comment (omeraplak) [inline examples/fineFoods/admin/antd/src/authProvider.ts:11]:
```suggestion
            localStorage.setItem(TOKEN_KEY, `${email}-${password}`);
            return Promise.resolve();
```

**Review by omeraplak — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**